backend/tests/test_services.py

```
@@ -151,3 +156,42 @@ def test_rag_embedding_failure_falls_back_to_lexical():
+
+
+def test_intent_gate_logs_when_llm_fails(caplog):
+    class BrokenLLM:
+        def is_enabled(self):
+            return True
+
+        def classify(self, _query: str):
+            raise RuntimeError('boom')
+
+    gate = IntentGateService(llm_service=BrokenLLM())
+    query = 'à¸­à¸¢à¸²à¸à¹„à¸›à¹€à¸—à¸µà¹ˆà¸¢à¸§à¸™à¹‰à¸³à¸•à¸'
```

Ruff (0.15.12)

[warning] String contains ambiguous `¸` (CEDILLA). Did you mean `,` (COMMA)?

(RUF001)